packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Models/Types/EnumTypeProvider.cs

```diff
@@ -0,0 +1,386 @@
+
+        public EnumTypeProvider(InputEnumType input, SourceInputModel? sourceInputModel) : base(sourceInputModel)
+        {
+            _allowedValues = input.AllowedValues;
+            _deprecated = input.Deprecated;
+
+            if (input.Accessibility == "internal")
+            {
+                DeclarationModifiers = TypeSignatureModifiers.Internal;
+            }
+            else
+            {
+                DeclarationModifiers = TypeSignatureModifiers.Public;
+            }
+            IsAccessibilityOverridden = input.Accessibility != null;
+
+            var isExtensible = input.IsExtensible;
+            if (ExistingType != null)
+            {
+                isExtensible = ExistingType.TypeKind switch
+                {
+                    TypeKind.Enum => false,
+                    TypeKind.Struct => true,
+                    _ => throw new InvalidOperationException(
+                        $"{ExistingType.ToDisplayString()} cannot be mapped to enum," +
+                        $" expected enum or struct got {ExistingType.TypeKind}")
+                };
+
+                _typeMapping = sourceInputModel?.CreateForModel(ExistingType);
+            }
+
+            if (isExtensible)
+            {
+                // extensible enums are implemented by readonly structs
+                DeclarationModifiers |= TypeSignatureModifiers.Partial | TypeSignatureModifiers.ReadOnly;
+            }
+
+            Name = input.Name.ToCleanName();
+            Namespace = GetDefaultModelNamespace(CodeModelPlugin.Instance.Configuration.Namespace);
+            IsExtensible = isExtensible;
+            ValueType = CodeModelPlugin.Instance.TypeFactory.CreateCSharpType(input.EnumValueType);
+            IsStringValueType = ValueType.Equals(typeof(string));
+            IsIntValueType = ValueType.Equals(typeof(int)) || ValueType.Equals(typeof(long));
+            IsFloatValueType = ValueType.Equals(typeof(float)) || ValueType.Equals(typeof(double));
+            IsNumericValueType = IsIntValueType || IsFloatValueType;
+
+            Description = input.Description;
+
+            _valueField = new FieldDeclaration(FieldModifiers.Private | FieldModifiers.ReadOnly, ValueType, "_value");
+            SerializationMethodName = IsStringValueType && IsExtensible ? nameof(object.ToString) : $"ToSerial{ValueType.Name.FirstCharToUpperCase()}";
+            DeserializationMethodName = $"To{Name}";
+
+            if (!IsExtensible)
+            {
+                Serialization = new EnumTypeSerializationProvider(this, sourceInputModel);
+            }
+        }
+
+        private readonly FieldDeclaration _valueField;
+
+        public CSharpType ValueType { get; }
+        public bool IsExtensible { get; }
+        public bool IsIntValueType { get; }
+        public bool IsFloatValueType { get; }
+        public bool IsStringValueType { get; }
+        public bool IsNumericValueType { get; }
+        public string? Description { get; }
```

Comment:
Are these properies needed on the public api surface?

https://apiview.dev/Assemblies/Review/6a1218ecd68c45a085d11e388ec86fa5/42a99366e43145aeab9fdf09baea8b61?diffRevisionId=dea184eb37d24ce59883b5aee3107ca9&doc=False&diffOnly=True
and
https://github.com/microsoft/typespec/pull/3313/files#diff-78f494ac2e1b834a186f094dcddb2c03567227c8d857a8fa7586c426fa86a473R90

It seems the writer shouldn't need any of this so it might all be private information?

Reply:
yeah these could be internal

Reply:
changed to internal